### DISCUSSION TITLE
feat: improve search (which is really poor right now)

### DISCUSSION THREAD

**Discussion by rubenfiszel** [scalar/scalar#1871] [Ideas]
One of the strength of scalar is the search function but the search work really badly in my experience.

For instance, typing: `list jobs` in https://app.windmill.dev/openapi.html#/

show 
![image](https://github.com/scalar/scalar/assets/275584/c5cd22a3-893b-4661-9df0-72ec135ec02f)

when there are actual methods that are `list all jobs`, `list queued jobs`, `list completed jobs` 
![image](https://github.com/scalar/scalar/assets/275584/ff8f8578-4d62-41aa-8c4f-4a55df9fce4c)

I'm almost sufficiently annoyed with it that I've considered doing a PR :D 

There are lots of great fuzzy search library out there, I recommend: https://github.com/leeoniya/uFuzzy

**Comment by marclave** [comment]
hey @rubenfiszel, appreciate the discussion and we should absolutely improve this search! let's see what we can get done today :)

**Reply by marclave** [reply]
some added context we are using fuzzy search :) https://www.fusejs.io/ but yeah we arent configuring it properly it seems, investigating!

**Comment by phamhieu** [comment]
Any updates on this issue? Supabase uses Scalar to power our public API docs, and overall the experience has been great. Thank you for the awesome work.

That said, we've started receiving quite a few complaints about the search experience, especially as the number of endpoints has grown. Users are finding it increasingly difficult to locate the most relevant results.

Here’s one example of recent feedback:

> Every time I search for something, the results aren’t great. For example, when I search for ‘logs’, the actual logs endpoint shows up third. I’d expect it to be the top result.

![Screenshot 2025-06-25 at 17 21 12](https://github.com/user-attachments/assets/db163862-259c-4ae6-8f5f-375f3bce9f5c)

We’re running into the same issue internally as well. Just wanted to surface this again in case there’s anything we can do on our end, or if improvements are planned.

**Reply by nicodand** [reply]
We are experiencing the same issue. The main pain point for us is that you can only search for endpoint titles and not paths.

**Reply by tien-truongn-otsv** [reply]
This issue due to the default options of fusejs.
I recommended allow passing custom Fuse.js fuzzy search configuration in this [Discussion](https://github.com/scalar/scalar/discussions/8703)

**Reply by hanspagel** [reply]
we don't use the default options?

**Reply by tien-truongn-otsv** [reply]
I noticed that Fuse.js in [useSearch](https://github.com/scalar/scalar/blob/main/packages/api-client/src/components/Search/useSearch.ts) is initialized with only the keys option configured, while other search behavior options (e.g. threshold, includeMatches, minMatchCharLength) are left at their defaults.

A few suggestions to improve the search experience:
   - Enable includeMatches: true to get highlighted/matched text ranges, which can be used to highlight matching parts in the search result popup
   -  Consider exposing a prop or option to allow passing custom Fuse.js configuration for more flexibility                                                                                                       

This would make it easier to display richer search results (e.g. bolding the matched text in titles/descriptions).

**Comment by hanspagel** [comment]
dramatically improved over the last 2 years already :)) open for further improvements
